libnavigation-ui/src/main/java/com/mapbox/navigation/ui/routealert/RouteAlertToll.kt

```diff
@@ -0,0 +1,121 @@
+package com.mapbox.navigation.ui.routealert
+
+import androidx.core.content.ContextCompat
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.style.expressions.Expression
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import com.mapbox.navigation.base.trip.model.alert.RouteAlert
+import com.mapbox.navigation.base.trip.model.alert.TollCollectionAlert
+import com.mapbox.navigation.ui.R
+import com.mapbox.navigation.ui.routealert.MapboxRouteAlertsDisplayer.Companion.getMapboxRouteAlertSymbolLayerProperties
+
+/**
+ * Handle [TollCollectionAlert] to add an icon on the map to indicate where a toll is presented.
+ *
+ * @param routeAlertViewOptions the options to build a route alert toll view
+ */
+
+class RouteAlertToll(
+    private var routeAlertViewOptions: RouteAlertViewOptions
+) {
+    private var tollAlertModels: List<RouteAlertModelToll> = emptyList()
+    private lateinit var tollCollectionsSource: GeoJsonSource
+    private lateinit var tollCollectionsLayer: SymbolLayer
+
+    private fun initialize() {
+        tollCollectionsSource = GeoJsonSource(MAPBOX_TOLL_COLLECTIONS_SOURCE)
+        tollCollectionsLayer = SymbolLayer(
+            MAPBOX_TOLL_COLLECTIONS_LAYER,
+            MAPBOX_TOLL_COLLECTIONS_SOURCE
+        )
+            .withProperties(
+                *routeAlertViewOptions.properties.let {
+                    if (it.isEmpty()) {
+                        getMapboxRouteAlertSymbolLayerProperties()
+                    } else {
+                        it
+                    }
+                },
+                PropertyFactory.iconImage(MAPBOX_TOLL_COLLECTIONS_IMAGE_PROPERTY_ID),
+                PropertyFactory.textField(
+                    Expression.get(Expression.literal(MAPBOX_TOLL_COLLECTIONS_TEXT_PROPERTY_ID))
+                )
+            )
+    }
+
+    /**
+     * When [Style] changes, re-add the toll route alerts to the new style.
+     *
+     * @param style the latest [Style]
+     */
+    fun onStyleLoaded(style: Style) {
+        initialize()
+
+        style.apply {
+            addImage(
+                MAPBOX_TOLL_COLLECTIONS_IMAGE_PROPERTY_ID,
+                routeAlertViewOptions.drawable
+                    ?: ContextCompat.getDrawable(
+                        routeAlertViewOptions.context,
+                        R.drawable.mapbox_ic_route_alert_toll
+                    )!!
+            )
+            addSource(tollCollectionsSource)
+            addLayer(tollCollectionsLayer)
```

Comment:
As per Maps SDK docs, when adding layers/sources to the map they have to be "newly created", which means that we should recreate them in `onStyleLoaded` as well.